### PR TITLE
added landsat orphans cleanup script

### PR DESCRIPTION
**Pull request by NikitaGandhi**
- added bash script for landsat orphans archive and cleanup. Not sure it's a best but it works.

### CODE REVIEW

File: scripts/misc/cleanup_landsat_orphans.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    # archive dataset and derive datasets
+    datacube dataset archive --archive-derived --dry-run $dataset_id
+
+    # s3 cleanup
+    for derived_location in $derived_locations; do
+      derived_path=`echo $derived_location | tr -d '"' | sed 's![^/]*$!!'`
+      echo "derived_path: $derived_path"
+      echo "remove derived scene here!"
+      # aws s3 rm $derived_path --recursive --dry-run
```
Comment (dominicschaff) [inline scripts/misc/cleanup_landsat_orphans.sh]: I am guessing these should not be commented out

Comment (NikitaGandhi) [reply scripts/misc/cleanup_landsat_orphans.sh]: yes! Just to avoid accident while I was testing a script.

**Comment by alexgleith** [comment]
I think we need to have a meeting and include Fang and Chad to check our logic.

And I think we need to take care about archiving children, as some child products include the GeoMAD, which we do NOT want to delete or archive, because we'd need to re-do the missing scenes then... well, I don't think we want to do this.

**Comment by NikitaGandhi** [comment]
> I think we need to have a meeting and include Fang and Chad to check our logic.
> 
> And I think we need to take care about archiving children, as some child products include the GeoMAD, which we do NOT want to delete or archive, because we'd need to re-do the missing scenes then... well, I don't think we want to do this.

@alexgleith I made suggested changes. Let me know if it right or still missing something.

**Comment by alexgleith** [comment]
Do you think it makes better sense to do the ARCHIVING and DELETING in two different steps?

So we have an archive_orphans script and a delete_archived script. Might be a bit safer, and possible to run again.

**Comment by NikitaGandhi** [comment]
> Do you think it makes better sense to do the ARCHIVING and DELETING in two different steps?
> 
> So we have an archive_orphans script and a delete_archived script. Might be a bit safer, and possible to run again.

@alexgleith In my current approach, the steps follow as below:
- get orphan dataset-id using s3-uri
- collect derived of orphan dataset - ids and s3-locations 
- archive derived and delete derive scene from s3
- then archive orphan and s3 cleanup

If we want to separate it into two scripts, then

archive_orphans script:
- read orphan report
- get orphan dataset-id using s3-uri
- collect derived of orphan dataset - ids and locations(needed for delete_archived script report)
- archive derived
- archive orphan
- generate delete_archive report that contains s3 paths for orphans and derived

delete_archived script:
- read delete_archive report
- go through each line and cleanup scene from s3

Let me know if this sound reasonable or can be simplified...

**Comment by NikitaGandhi** [comment]
Raised a new PR - #86 with suggested changes.

**Pull request closed without merging**